tests/nosetests/pyanaconda_tests/module_payload_test.py

```diff
@@ -0,0 +1,275 @@
+        ks_in = """
+        %packages
+        package
+        @group
+        @module:10
+        @module2:1.5/server
+        @^environment
+        %end
+        """
+        ks_out = """
+        %packages
+        @^environment
+        @group
+        @module2:1.5/server
+        @module:10
+        package
+
+        %end
+        """
+        self._test_kickstart(ks_in, ks_out)
+
+    def packages_section_complex_kickstart_test(self):
+        """Test the packages section with duplicates."""
+        ks_in = """
+        %packages
+        @^environment1
+        package1
+        @group1
+        package2
+
+        # Only this environment will stay (last specified wins)
+        @^environment2
+        @group2
+
+        # duplicates
+        package2
+        @group2
+
+        # modules
+        @module:4
+        @module:3.5/server
+
+        %end
+        """
+        # The last specified environment wins, you can't specify two environments
+        # Same package or group specified twice will be deduplicated
+        ks_out = """
+        %packages
+        @^environment2
+        @group1
+        @group2
+        @module:3.5/server
+        @module:4
+        package1
+        package2
+
+        %end
+        """
+        self._test_kickstart(ks_in, ks_out)
+
+    def packages_section_with_attribute_kickstart_test(self):
+        """Test the packages section with attribute."""
+        ks_in = """
+        %packages --nocore
+        %end
+        """
+        ks_out = """
+        %packages --nocore
+
+        %end
+        """
+        self._test_kickstart(ks_in, ks_out)
+
+    def packages_section_multiple_attributes_kickstart_test(self):
+        """Test the packages section with multiple attributes."""
+        ks_in = """
+        %packages --nocore --multilib --instLangs en_US.UTF-8
+
+        %end
+        """
+        ks_out = """
+        %packages --nocore --instLangs=en_US.UTF-8 --multilib
+
+        %end
+        """
+        self._test_kickstart(ks_in, ks_out)
+
+    def packages_section_excludes_kickstart_test(self):
+        """Test the packages section with excludes."""
+        ks_in = """
+        %packages
+        -vim
+        %end
+        """
+        ks_out = """
+        %packages
+        -vim
+
+        %end
+        """
+        self._test_kickstart(ks_in, ks_out)
+
+    def packages_section_complex_exclude_kickstart_test(self):
+        """Test the packages section with complex exclude example."""
+        ks_in = """
+        %packages --nocore
+        @^environment1
+        @group1
+        package1
+        -package2
+        -@group2
+        @group3
+        package3
+        %end
+        """
+        ks_out = """
+        %packages --nocore
+        @^environment1
+        @group1
+        @group3
+        package1
+        package3
+        -@group2
+        -package2
+
+        %end
+        """
+        self._test_kickstart(ks_in, ks_out)
+
+    def no_core_group_properties_test(self):
```

Comment:
I recommend moving the tests for the Packages interface to a new file.

Reply:
I have that in my mind but in another test later when there will be more API and not just for %packages section.

Reply:
In other words will be done in a separate PR. Right now, we don't have anything else so this can be "marked" as payload module tests.

Reply:
Ok.